### ISSUE TITLE
USB_DEVICE_AND_INTERFACE_INFO" requires 5 arguments, but only 4 given

### ISSUE THREAD

**Issue by mrmodolo** [morrownr/8821cu#1]
Hi!

I followed your instructions but the error below happened!

```
/home/modolo/src/8821cu/os_dep/linux/usb_intf.c:268:64: error: macro "USB_DEVICE_AND_INTERFACE_INFO" requires 5 arguments, but only 4 given
  268 |  {USB_DEVICE_AND_INTERFACE_INFO(0x2001, 0x331d 0xff, 0xff, 0xff), .driver_info = RTL8821C}, /* D-Link - DWA-171C */
      |                                                                ^
In file included from /home/modolo/src/8821cu/include/osdep_service_linux.h:111,
                 from /home/modolo/src/8821cu/include/osdep_service.h:53,
                 from /home/modolo/src/8821cu/include/drv_types.h:27,
                 from /home/modolo/src/8821cu/os_dep/linux/usb_intf.c:17:
./include/linux/usb.h:1057: note: macro "USB_DEVICE_AND_INTERFACE_INFO" defined here
 1057 | #define USB_DEVICE_AND_INTERFACE_INFO(vend, prod, cl, sc, pr) \
      | 
/home/modolo/src/8821cu/os_dep/linux/usb_intf.c:268:3: error: ‘USB_DEVICE_AND_INTERFACE_INFO’ undeclared here (not in a function)
  268 |  {USB_DEVICE_AND_INTERFACE_INFO(0x2001, 0x331d 0xff, 0xff, 0xff), .driver_info = RTL8821C}, /* D-Link - DWA-171C */
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [scripts/Makefile.build:275: /home/modolo/src/8821cu/os_dep/linux/usb_intf.o] Erro 1
make[1]: *** [Makefile:1757: /home/modolo/src/8821cu] Erro 2
make[1]: Saindo do diretório '/usr/src/linux-headers-5.4.0-52-generic'
make: *** [Makefile:2349: modules] Erro 2
```
lsb_release -a
LSB Version:	core-11.1.0ubuntu2-noarch:printing-11.1.0ubuntu2-noarch:security-11.1.0ubuntu2-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal

Thanks,
Marcelo Módolo

**Comment by morrownr** [comment]
Thanks for the report. This was my fault. I was adding a couple of device IDs and I forgot a comma. I have added the comma and tested. 

Please use the ./dkms-remove.sh script to remove any installed components and then delete the package you have. Then download the updated driver and try again.

Humble apologies for this error. Please let me know if it works this time and if there are any improvements I can make to the README.

**Comment by morrownr** [comment]
Mr Modolo, if your wifi adapter has an LED light, can I get you to test all 3 options and report back whether this is working properly? My test adapter does not have an LED light so I cannot test this.

Thanks

**Comment by mrmodolo** [comment]
Hi!

It worked really well!

![off](https://user-images.githubusercontent.com/1923174/97873653-b5a5b300-1cf6-11eb-914a-f83c8655100e.png)
![on](https://user-images.githubusercontent.com/1923174/97873654-b63e4980-1cf6-11eb-8c83-3dae8caa0c85.png)

```
systool -vm 8821cu | grep -E '(rtw_drv_log_level|rtw_led_ctrl|rtw_btcoex_enable|rtw_country_code)'
    rtw_btcoex_enable   = "0"
    rtw_country_code    = "BR"
    rtw_drv_log_level   = "0"
    rtw_led_ctrl        = "1"
```
I made some changes (rtw_btcoex_enable=0) to the file /etc/modprobe.d/8821cu.conf because I have been having problems with the MX Anywere 25 mouse. But this has nothing to do with your driver, I used this configuration before and I'm testing to see if it will solve the problem!

Congratulations on delivering this driver to the community, the build did not present any errors or alerts!

Thanks,

Marcelo Módolo
